plugins/outputs/dynatrace/dynatrace_test.go

```diff
@@ -515,3 +515,73 @@ func TestSendCounterMetricWithoutTags(t *testing.T) {
 	err = d.Write(metrics)
 	require.NoError(t, err)
 }
+
+var warnfCalledTimes int
+
+type loggerStub struct {
+	testutil.Logger
+}
+
+func (l loggerStub) Warnf(format string, args ...interface{}) {
+	warnfCalledTimes++
+}
+
+func TestSendUnsupportedMetric(t *testing.T) {
+	warnfCalledTimes = 0
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		t.Fatal("should not export because the only metric is an invalid type")
+	}))
+	defer ts.Close()
+
+	d := &Dynatrace{}
+
+	logStub := loggerStub{}
+
+	d.URL = ts.URL
+	d.APIToken = "123"
+	d.Log = logStub
+	err := d.Init()
+	require.NoError(t, err)
+	err = d.Connect()
+	require.NoError(t, err)
+
+	// Init metrics
+
+	m1 := metric.New(
+		"mymeasurement",
+		map[string]string{},
+		map[string]interface{}{"1": "unsupported_type"},
```

Comment:
```suggestion
		map[string]interface{}{"metric1": "unsupported_type"},
```
Otherwise it looks at first like "mymeasurement" was the metric name and "1" the stringified value.